### PR TITLE
ferium: init at 4.1.1

### PR DESCRIPTION
**Pull request by leo60228**
###### Description of changes

Packaged [ferium](https://github.com/theRookieCoder/ferium), a CLI Minecraft mod manager.
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Mindavi — APPROVED** [review]
Diff LGTM, didn't test.

**Comment by Mindavi** [comment]
Oh, see upstream issue. Patch can be removed with newest release.

**Comment by Mindavi** [comment]
Might wanna disable darwin for now: https://logs.nix.ci/?key=nixos/nixpkgs.173201&attempt_id=32174c2a-65e6-42a6-9c23-ded86d81f98d.

E.g. by setting meta.platforms = lib.platforms.linux. Unless you plan to maintain darwin support or if someone volunteers.

**Comment by leo60228** [comment]
I don't plan to maintain Darwin but I don't see changes being necessary in the future so I'll just try and get it working based on CI for now.

**Comment by leo60228** [comment]
It seems like with that change macOS builds.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -31678,6 +31678,8 @@ with pkgs;
 
   fairymax = callPackage ../games/fairymax { };
 
+  ferium = callPackage ../games/ferium { };
```
Comment (06kellyjac) [inline pkgs/top-level/all-packages.nix:31681]:
```suggestion
  ferium = callPackage ../games/ferium {
  	inherit (darwin.apple_sdk.frameworks) Security;
  };
```

File: pkgs/games/ferium/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  version = "4.0.0";
+
+  src = fetchFromGitHub {
+    owner = "gorilla-devs";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "x781dQpG0N6pcEuF8C1R8g6cUNIevOTUMG30ruJY9+Y=";
+  };
+
+  buildInputs = lib.optionals stdenv.isDarwin [ darwin.Security ];
```
Comment (06kellyjac) [inline pkgs/games/ferium/default.nix:14]:
```suggestion
{ lib, stdenv, fetchFromGitHub, rustPlatform, Security }:

rustPlatform.buildRustPackage rec {
  pname = "ferium";
  version = "4.0.0";

  src = fetchFromGitHub {
    owner = "gorilla-devs";
    repo = pname;
    rev = "v${version}";
    sha256 = "x781dQpG0N6pcEuF8C1R8g6cUNIevOTUMG30ruJY9+Y=";
  };

  buildInputs = lib.optionals stdenv.isDarwin [ Security ];
```

**Review by 06kellyjac — COMMENTED** [review]
This is how we typically pass in darwin frameworks

Also the PR title needs to reflect the commit

**Comment by leo60228** [comment]
Ugh, I remembered that review and thought I followed up on it... will do that now.

**Comment by leo60228** [comment]
Addressed the review. It currently does not build as the latest version of ferium requires Rust 1.61.

**Comment by imsofi** [comment]
You can patch out the 1.61 requirement rather easily.

You can check out how im doing it here: https://github.com/imsofi/nur-pkgs/blob/main/pkgs/ferium

**Comment by leo60228** [comment]
Well, Rust 1.61 is on master now, so... I'll just rebase.

**Comment by happysalada** [comment]
CI is happy, diff looks good, thank you!

**Pull request merged**